src/ui/common/src/components/pages/workflows/index.tsx

```diff
@@ -66,52 +67,58 @@ const WorkflowsPage: React.FC<Props> = ({ user, Layout = DefaultLayout }) => {
   /**
    * Iterate through workflows array and map each element to a WorkflowTableRow object.
    */
-  const workflowElements: PaginatedSearchTableRow[] = workflows.map((value) => {
-    const engine = value.engine;
-
-    let metrics = [];
-    if (value?.metrics) {
-      metrics = value.metrics.map((metric) => {
-        return {
-          metricId: metric.id,
-          name: metric.name,
-          value: metric.result?.content_serialized ?? '',
-          status: metric.result?.exec_state?.status ?? ExecutionStatus.Unknown,
-        };
-      });
-    }
+  const workflowElements: PaginatedSearchTableRow[] = workflows.map(
+    (workflow) => {
+      const engines = reduceEngineTypes(
+        workflow.engine,
+        workflow.operator_engines.filter((x) => !!x)
+      );
+
+      let metrics = [];
+      if (workflow?.metrics) {
+        metrics = workflow.metrics.map((metric) => {
+          return {
+            metricId: metric.id,
+            name: metric.name,
+            value: metric.result?.content_serialized ?? '',
+            status:
+              metric.result?.exec_state?.status ?? ExecutionStatus.Unknown,
+          };
+        });
+      }
 
-    let checks = [];
-    if (value.checks) {
-      checks = value.checks.map((check) => {
-        const value =
-          check.result?.exec_state.status === 'succeeded' ? 'True' : 'False';
-
-        return {
-          checkId: check.id,
-          name: check.name,
-          level: check.spec?.check?.level ?? CheckLevel.Warning,
-          timestamp: check.result?.exec_state?.timestamps?.finished_at ?? '',
-          value,
-          status: check.result?.exec_state?.status ?? ExecutionStatus.Unknown,
-        };
-      });
-    }
+      let checks = [];
+      if (workflow.checks) {
+        checks = workflow.checks.map((check) => {
+          const value =
+            check.result?.exec_state.status === 'succeeded' ? 'True' : 'False';
+
+          return {
+            checkId: check.id,
+            name: check.name,
+            level: check.spec?.check?.level ?? CheckLevel.Warning,
```

Comment:
Are we defaulting to warning level if there is no level defined for the check?

I find that kind of odd, figured most people would go to error, but I suppose this is to keep the workflow from failing if the check fails?

Reply:
I think this is more a safeguard to show something and the UI doesn't crash. There shouldn't be practical use case where this value is empty. If this actually happens the backend is likely to complain and more places in UI will error.